### PR TITLE
Refactor code for fewer tests with added description

### PR DESCRIPTION
**Pull request by mattiahj**
I want to simplify our tests and sparql query. I also want to add descriptions to our tests so we know what they're meant to be testing.

### CODE REVIEW

File: metadata-transformer/src/test/java/no/unit/nva/metadata/service/MetadataServiceTest.java [hunk 1]
```diff
@@ -56,18 +60,19 @@
 
     private WireMockServer wireMockServer;
 
-    @ParameterizedTest
+    @ParameterizedTest(name = "#{index} - {0}")
```
Comment (mattiahj) [inline metadata-transformer/src/test/java/no/unit/nva/metadata/service/MetadataServiceTest.java:63]: If people have naming schemes they prefer, please suggest them.

File: metadata-transformer/src/main/java/no/unit/nva/metadata/NoLowercaseDCPrefixInMetadataFix.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
+package no.unit.nva.metadata;
+
+import org.apache.any23.validator.*;
+import org.w3c.dom.Node;
+
+import java.util.List;
```
Comment (brinxmat) [inline metadata-transformer/src/main/java/no/unit/nva/metadata/NoLowercaseDCPrefixInMetadataFix.java:6]: I guess I would understand this more intuitively when reading the code (if I understood your code correctly). I can't verify the functionality as there is no test.

```
package no.unit.nva.metadata;

import org.apache.any23.validator.DOMDocument;
import org.apache.any23.validator.Fix;
import org.apache.any23.validator.Rule;
import org.apache.any23.validator.RuleContext;
import org.w3c.dom.Node;



public class NoLowercaseDCPrefixInMetadataFix extends NoLowercase implements Fix {
    public static final String UPPERCASE_DC = "DC.";
    public static final String NO_LOWERCASE_DC_PREFIX_FIX = "no-lowercase-dc-prefix-in-metadata-fix";
    public static final int PREFIX_END = 3;

    @Override
    public String getHRName() {
        return NO_LOWERCASE_DC_PREFIX_FIX;
    }

    @Override
    public void execute(Rule rule, RuleContext context, DOMDocument document) {
        document.getNodes(HTML_META_TAGS_XPATH).forEach(this::fix);
    }

    private void fix(Node node) {
        Node nameNode = getMetaTagNameAttributeValue(node);
        if (containsLowerCaseDcPrefix(nameNode)) {
            nameNode.setTextContent(UPPERCASE_DC + nameNode.getTextContent().substring(PREFIX_END));
        }
    }
}
```

Comment (brinxmat) [reply metadata-transformer/src/main/java/no/unit/nva/metadata/NoLowercaseDCPrefixInMetadataFix.java:6]: And also

```
package no.unit.nva.metadata;

import org.w3c.dom.Node;

import static java.util.Objects.nonNull;

public class NoLowercase {
    public static final String HTML_META_TAGS_XPATH = "/HTML/HEAD/META";
    public static final String META_TAG_NAME_ATTRIBUTE = "name";
    public static final String LOWERCASE_DC = "dc.";

    protected Node getMetaTagNameAttributeValue(Node node) {
        return node.getAttributes().getNamedItem(META_TAG_NAME_ATTRIBUTE);
    }

    protected boolean containsLowerCaseDcPrefix(Node nameNode) {
        return nonNull(nameNode) && nameNode.getTextContent().startsWith(LOWERCASE_DC);
    }
}
```

File: metadata-transformer/src/main/java/no/unit/nva/metadata/NoLowercaseDCPrefixInMetadataRule.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
+package no.unit.nva.metadata;
+
+import org.apache.any23.validator.*;
+import org.w3c.dom.Node;
+
+import java.util.List;
+
+public class NoLowercaseDCPrefixInMetadataRule implements Rule {
```
[the remark this answers is not in the document]
Comment (brinxmat) [reply metadata-transformer/src/main/java/no/unit/nva/metadata/NoLowercaseDCPrefixInMetadataRule.java:8]: Both of the fixes assume that they are applied at some point, in order  to do this, the TranslatorService::loadMetadataFromUri needs to be adapted.

File: metadata-transformer/src/test/java/no/unit/nva/metadata/service/MetadataServiceTest.java [hunk 4]
```diff
@@ -138,19 +143,19 @@ private static CreatePublicationRequest requestWithContributorName(String name)
...
-                        DC_DATE_UPPER_CASE, dateString),
-                        request),
-                generateMetadataHtml(Map.of(
-                        DC_DATE, year),
-                        yearOnlyRequest),
-                generateMetadataHtml(Map.of(
-                        DC_DATE_UPPER_CASE, year),
-                        yearOnlyRequest)
-                );
+            generateTestWithMetadataHtml("dc.date with year, month and day map to request",
```
Comment (brinxmat) [inline metadata-transformer/src/test/java/no/unit/nva/metadata/service/MetadataServiceTest.java:146]: …request PublicationDate

Comment (brinxmat) [reply metadata-transformer/src/test/java/no/unit/nva/metadata/service/MetadataServiceTest.java:146]: On all of these…I think it is better to be explicit

File: metadata-transformer/src/test/java/no/unit/nva/metadata/service/MetadataServiceTest.java [hunk 5]
```diff
@@ -168,20 +173,21 @@ private static CreatePublicationRequest createRequestWithDate(PublicationDate da
...
-                generateMetadataHtml(Map.of(
-                        DC_TITLE_UPPER_CASE, title),
-                        request),
-                generateMetadataHtml(Map.of(
-                        DC_TITLE, title,
-                        DC_TITLE_UPPER_CASE, title,
-                        CITATION_TITLE, longerTitle),
-                        longerTitleRequest)
-                );
+            generateTestWithMetadataHtml("dc.title map to request", Map.of(
```
Comment (brinxmat) [inline metadata-transformer/src/test/java/no/unit/nva/metadata/service/MetadataServiceTest.java:177]: …request mainTitle on all of these

File: metadata-transformer/src/test/java/no/unit/nva/metadata/service/MetadataServiceTest.java [hunk 6]
```diff
@@ -199,11 +205,15 @@ private static CreatePublicationRequest createRequestWithTitle(String title) {
         CreatePublicationRequest request = createRequestWithTags(List.of(subject, coverage));
 
         return Stream.of(
-                generateMetadataHtml(Map.of(
-                        DC_COVERAGE, coverage,
-                        DC_SUBJECT, subject),
-                        request)
-                );
+            generateTestWithMetadataHtml("dc.coverage and dc.subject map to request", Map.of(
```
Comment (brinxmat) [inline metadata-transformer/src/test/java/no/unit/nva/metadata/service/MetadataServiceTest.java:208]: …to whatever it is that these actually map to…I'm not even sure :D

File: metadata-transformer/src/test/java/no/unit/nva/metadata/service/MetadataServiceTest.java [hunk 7]
```diff
@@ -220,23 +230,25 @@ private static CreatePublicationRequest createRequestWithTags(List<String> tags)
...
-                        DC_DESCRIPTION, description),
-                        request),
-                generateMetadataHtml(Map.of(
-                        DC_DESCRIPTION, abstractString),
-                        abstractOnlyRequest),
-                generateMetadataHtml(Map.of(
-                        DCTERMS_ABSTRACT, abstractString),
-                        abstractOnlyRequest)
-                 );
+            generateTestWithMetadataHtml("dcterms.description and dc.description maps to request", Map.of(
```
Comment (brinxmat) [inline metadata-transformer/src/test/java/no/unit/nva/metadata/service/MetadataServiceTest.java:237]: …request Abstract when any abstract is missing?

File: metadata-transformer/src/test/java/no/unit/nva/metadata/service/MetadataServiceTest.java [hunk 8]
```diff
@@ -220,23 +230,25 @@ private static CreatePublicationRequest createRequestWithTags(List<String> tags)
...
+                DCTERMS_ABSTRACT, abstractString,
+                DC_DESCRIPTION, description),
+                request),
+            generateTestWithMetadataHtml("DCTERMS.description and DC.description maps to request", Map.of(
```
Comment (brinxmat) [inline metadata-transformer/src/test/java/no/unit/nva/metadata/service/MetadataServiceTest.java:241]: Ditto

File: metadata-transformer/src/test/java/no/unit/nva/metadata/service/MetadataServiceTest.java [hunk 9]
```diff
@@ -250,23 +262,23 @@ private static CreatePublicationRequest createRequestWithTags(List<String> tags)
...
-                generateMetadataHtml(Map.of(
-                        DC_TITLE, dummyTitle,
-                        DC_IDENTIFIER_UPPER_CASE, expectedDoi),
-                        request),
-                generateMetadataHtml(Map.of(
-                        DC_TITLE, dummyTitle,
-                        DC_IDENTIFIER, notADoi),
-                        emptyRequest)
-                );
+            generateTestWithMetadataHtml("dc.identifier with doi prefix map to request", Map.of(
```
Comment (brinxmat) [inline metadata-transformer/src/test/java/no/unit/nva/metadata/service/MetadataServiceTest.java:265]: …request...DOI?

File: metadata-transformer/src/test/java/no/unit/nva/metadata/service/MetadataServiceTest.java [hunk 10]
```diff
@@ -277,14 +289,17 @@ private static CreatePublicationRequest createRequestWithTags(List<String> tags)
...
         CreatePublicationRequest underterminedRequest = requestWithLanguage(URI.create(underterminedLanguage));
-
         return Stream.of(
-            generateMetadataHtml(Map.of(DC_LANGUAGE, language), request),
-            generateMetadataHtml(Map.of(DC_LANGUAGE_UPPER_CASE, uppercaseLanguage), request),
-            generateMetadataHtml(Map.of(DC_LANGUAGE, notALanguage), underterminedRequest),
-            generateMetadataHtml(Map.of(DC_LANGUAGE_UPPER_CASE, ""), underterminedRequest),
-            generateMetadataHtml(Map.of(DC_LANGUAGE, " "), underterminedRequest),
-            generateMetadataHtml(DC_LANGUAGE, null, underterminedRequest)
+            generateTestWithMetadataHtml("dc.language with BCP-47 code map to ISO639-3 code in request",
```
Comment (brinxmat) [inline metadata-transformer/src/test/java/no/unit/nva/metadata/service/MetadataServiceTest.java:293]: request mainLanguage?

File: metadata-transformer/src/main/java/no/unit/nva/metadata/NoLowercaseDCPrefixInMetadataRule.java [hunk 11]
```diff
@@ -0,0 +1,30 @@
+package no.unit.nva.metadata;
```
Comment (brinxmat) [inline metadata-transformer/src/main/java/no/unit/nva/metadata/NoLowercaseDCPrefixInMetadataRule.java:1]: This doesn't belong in this PR :)

**Review by brinxmat — CHANGES_REQUESTED** [review]

File: metadata-transformer/src/main/java/no/unit/nva/metadata/NoLowercaseDCPrefixInMetadataFix.java [hunk 12]
```diff
@@ -0,0 +1,25 @@
+package no.unit.nva.metadata;
```
[the remark this answers is not in the document]
Comment (mattiahj) [reply metadata-transformer/src/main/java/no/unit/nva/metadata/NoLowercaseDCPrefixInMetadataFix.java:1]: Honestly, this entire commit is no more than a sketch; like you said, more rewrites would be needed to apply this validation rule. I mostly wanted to know if this was what you wanted... Personally, I suspect it might be easier to apply this in MetadataConverter somehow without getting validator mixed into it. What do you think?

File: metadata-transformer/src/test/java/no/unit/nva/metadata/service/MetadataServiceTest.java (hunk 9 above)
Comment (mattiahj) [reply metadata-transformer/src/test/java/no/unit/nva/metadata/service/MetadataServiceTest.java:265]: I was thinking more.. "map to createRequest" maybe "to doi in createRequest" 
am I off base?

File: metadata-transformer/src/main/java/no/unit/nva/metadata/NoLowercaseDCPrefixInMetadataFix.java [hunk 13]
```diff
@@ -0,0 +1,25 @@
...
+    }
+
+    @Override
+    public void execute(Rule rule, RuleContext context, DOMDocument document) {
+        List<Node> metas = document.getNodes("/HTML/HEAD/META");
+        for (Node meta : metas) {
+            Node nameNode = meta.getAttributes().getNamedItem("name");
+            if (nameNode != null && nameNode.getTextContent().startsWith("dc.")) {
+                nameNode.setTextContent("DC." + nameNode.getTextContent().substring(3));
+            }
```
Comment (axthosarouris) [inline metadata-transformer/src/main/java/no/unit/nva/metadata/NoLowercaseDCPrefixInMetadataFix.java:22]: refactor to a method.

File: metadata-transformer/src/main/java/no/unit/nva/metadata/NoLowercaseDCPrefixInMetadataRule.java [hunk 14]
```diff
@@ -0,0 +1,30 @@
...
+    public String getHRName() {
+        return "no-lowercase-dc-prefix-in-metadata-rule";
+    }
+
+    @Override
+    public boolean applyOn(DOMDocument document, RuleContext<?> context,
+                           ValidationReportBuilder validationReportBuilder) {
+        List<Node> metas = document.getNodes("/HTML/HEAD/META");
+        for (Node meta : metas) {
+            Node nameNode = meta.getAttributes().getNamedItem("name");
```
Comment (axthosarouris) [inline metadata-transformer/src/main/java/no/unit/nva/metadata/NoLowercaseDCPrefixInMetadataRule.java:20]: I think what you are doing here is in pseudocode:
```
if( existsNodeWithLowerCaseDcPrefix(metas)){
reportError();
return true;
}

return false;
```

Comment (axthosarouris) [reply metadata-transformer/src/main/java/no/unit/nva/metadata/NoLowercaseDCPrefixInMetadataRule.java:20]: I think if you refactor the code it will describe the intention better.

File: metadata-transformer/src/test/java/no/unit/nva/metadata/service/MetadataServiceTest.java [hunk 15]
```diff
@@ -56,18 +60,19 @@
...
+        "provideMetadataForAbstract",
+        "provideMetadataForTitle",
+        "provideMetadataForDate",
+        "provideMetadataForContributors",
+        "provideMetadataForIdentifier",
+        "provideMetadataForLanguage"
     })
-    public void getCreatePublicationParsesHtmlAndReturnsMetadata(String html, CreatePublicationRequest expectedRequest)
-            throws IOException {
+    public void getCreatePublicationParsesHtmlAndReturnsMetadata(String testDescription, String html,
```
Comment (axthosarouris) [inline metadata-transformer/src/test/java/no/unit/nva/metadata/service/MetadataServiceTest.java:73]: testDescription is not being used.

File: metadata-transformer/src/main/java/no/unit/nva/metadata/NoLowercaseDCPrefixInMetadataFix.java (hunk 2 above)
Comment (mattiahj) [reply metadata-transformer/src/main/java/no/unit/nva/metadata/NoLowercaseDCPrefixInMetadataFix.java:6]: Hmm. I would not. I don't intuitively understand the difference between "execute" and "fix". Having both is confusing. Also it breaks with Apaches example (and arguably their interface) of how these classes are meant to be implemented.

File: metadata-transformer/src/test/java/no/unit/nva/metadata/service/MetadataServiceTest.java [hunk 16]
```diff
@@ -87,19 +92,19 @@ public void getCreatePublicationParsesHtmlAndReturnsMetadata(String html, Create
...
-                        DC_CONTRIBUTOR_UPPER_CASE, name),
-                        request),
-                generateMetadataHtml(Map.of(
-                        DC_CREATOR, name),
-                        request),
-                generateMetadataHtml(Map.of(
-                        DC_CREATOR_UPPER_CASE, name),
-                        request)
-                );
+            generateTestWithMetadataHtml("dc.contributor map to contributor name in request",
```
Comment (axthosarouris) [inline metadata-transformer/src/test/java/no/unit/nva/metadata/service/MetadataServiceTest.java:95]: It seems that the intention of of these tests is to test whether non-capital and capital prefixes are accepted. The intention of each line can be understood only by comparing it to another line in the test. It would be easier if they were self contained. Example: LOWER_CASE_DC_PREFIX_CONTRIBUTOR_MAP, UPPER_CASE_DC_PREFIX_CONTRIBUTOR_MAP

File: metadata-transformer/src/main/java/no/unit/nva/metadata/NoLowercaseDCPrefixInMetadataRule.java [hunk 17]
```diff
@@ -0,0 +1,30 @@
...
+            if (nameNode != null && nameNode.getTextContent().startsWith("dc.")) {
+                validationReportBuilder.reportIssue(ValidationReport.IssueLevel.ERROR,
```
Comment (axthosarouris) [inline metadata-transformer/src/main/java/no/unit/nva/metadata/NoLowercaseDCPrefixInMetadataRule.java:22]: is there a test for the validation report?

**Review by axthosarouris — CHANGES_REQUESTED** [review]
Since many tests are merged together to a single parametrized test, I would suggest that we name each string instance with the case that is being tested. Otherwise in a later refactoring it will be impossible to understand what is the role of each string instance. An alternative is to break the parametrized test in single tests since some cases also produce report messages.

File: metadata-transformer/src/main/java/no/unit/nva/metadata/NoLowercaseDCPrefixInMetadataFix.java (hunk 12 above)
[the remark this answers is not in the document]
Comment (mattiahj) [reply metadata-transformer/src/main/java/no/unit/nva/metadata/NoLowercaseDCPrefixInMetadataFix.java:1]: I had hoped that we could find some solution that would allow us to do away with the lowercase/uppercase tests, at that point it makes sense to include in this PR.

File: metadata-transformer/src/main/java/no/unit/nva/metadata/NoLowercaseDCPrefixInMetadataFix.java (hunk 2 above)
Comment (janbj) [reply metadata-transformer/src/main/java/no/unit/nva/metadata/NoLowercaseDCPrefixInMetadataFix.java:6]: the private method is better for readability, but the name could be improved and is specific to the implementation.

File: metadata-transformer/src/main/java/no/unit/nva/metadata/NoLowercaseDCPrefixInMetadataFix.java (hunk 12 above)
[the remark this answers is not in the document]
Comment (brinxmat) [reply metadata-transformer/src/main/java/no/unit/nva/metadata/NoLowercaseDCPrefixInMetadataFix.java:1]: Difficult for me as I was looking at the PR title

File: metadata-transformer/src/main/java/no/unit/nva/metadata/NoLowercaseDCPrefixInMetadataFix.java (hunk 2 above)
Comment (brinxmat) [reply metadata-transformer/src/main/java/no/unit/nva/metadata/NoLowercaseDCPrefixInMetadataFix.java:6]: PR on my PR…meta :D

File: metadata-transformer/src/test/java/no/unit/nva/metadata/service/MetadataServiceTest.java (hunk 9 above)
Comment (brinxmat) [reply metadata-transformer/src/test/java/no/unit/nva/metadata/service/MetadataServiceTest.java]: doi in createRequest, I think

File: metadata-transformer/src/main/java/no/unit/nva/metadata/NoLowercaseDCPrefixInMetadataRule.java (hunk 17 above)
Comment (mattiahj) [reply metadata-transformer/src/main/java/no/unit/nva/metadata/NoLowercaseDCPrefixInMetadataRule.java:22]: > Since many tests are merged together to a single parametrized test, I would suggest that we name each string instance with the case that is being tested. Otherwise in a later refactoring it will be impossible to understand what is the role of each string instance.

That is the intent of this PR.

File: metadata-transformer/src/main/java/no/unit/nva/metadata/NoLowercaseDCPrefixInMetadataFix.java [hunk 18]
```diff
@@ -0,0 +1,25 @@
+package no.unit.nva.metadata;
+
+import org.apache.any23.validator.*;
```
[the remark this answers is not in the document]
Comment (mattiahj) [reply metadata-transformer/src/main/java/no/unit/nva/metadata/NoLowercaseDCPrefixInMetadataFix.java:3]: I know this is your convention, and don't worry, I am, this was just a quick draft to get some feedback on the general idea.

I'm guessing you meant to say that it loads the whole package, but there's really no reason to worry. The compiler replaces the wildcard import with the specific classes used in the code. I believe the real rationale for not using wildcard imports is to avoid namespace conflicts and for developers to be able to read up top which classes are imported, but the first issue can be addressed and the second one is subjective; in fact Clean Code recommends using wildcard imports.

File: metadata-transformer/src/test/java/no/unit/nva/metadata/service/MetadataServiceTest.java (hunk 15 above)
Comment (mattiahj) [reply metadata-transformer/src/test/java/no/unit/nva/metadata/service/MetadataServiceTest.java:70]: It actually is; by the mapper. Without the correct number of arguments in the right position it gets confused. If there's a better way of solving this issue I'd welcome your suggestion.

File: metadata-transformer/src/main/java/no/unit/nva/metadata/NoLowercaseDCPrefixInMetadataRule.java (hunk 14 above)
Comment (mattiahj) [reply metadata-transformer/src/main/java/no/unit/nva/metadata/NoLowercaseDCPrefixInMetadataRule.java:20]: I think the intent is pretty clear, but then again I just wrote it, and it might not be as clear for a theoretical dev comes across this not knowing what the Any23 Validator is. Writing it out like you specified I don't actually find the code to be much more readable, we've just added a descriptive method name, and honestly, the class name should have gotten the point across. I'm not sure I'd consider this worth 5 more lines, and doubling the amount of return statements, but maybe that's just me. I'll mess about with it some more.

File: metadata-transformer/src/main/java/no/unit/nva/metadata/NoLowercaseDCPrefixInMetadataFix.java (hunk 12 above)
[the remark this answers is not in the document]
Comment (mattiahj) [reply metadata-transformer/src/main/java/no/unit/nva/metadata/NoLowercaseDCPrefixInMetadataFix.java]: Ah, sorry about that.

File: metadata-transformer/src/main/java/no/unit/nva/metadata/service/MetadataService.java [hunk 19]
```diff
@@ -73,23 +77,56 @@ private ByteArrayInputStream loadExtractedData() {
...
+            if (isSindiceDcOrDcTerms(statement)) {
+                model.add(valueFactory.createStatement(statement.getSubject(),
+                    toDcNamespace(statement.getPredicate(), valueFactory),
+                    statement.getObject()));
+            } else {
+                model.add(statement);
+            }
+        }
+        return model;
+    }
```
Comment (brinxmat) [inline metadata-transformer/src/main/java/no/unit/nva/metadata/service/MetadataService.java:111]:
```suggestion
   private Model fixUpSindiceDcAndDctermsToDcterms(RepositoryResult<Statement> statements) {
        ValueFactory valueFactory = SimpleValueFactory.getInstance();
        Model model = new TreeModel();
        while (statements.hasNext()) {
            @SuppressWarnings("PMD.CloseResource")
            Statement statement = statements.next();
            model.add(fixDcAndDcTermsStatement(valueFactory, statement));
        }
        return model;
    }

    private Statement fixDcAndDcTermsStatement(ValueFactory valueFactory, Statement statement) {
        if (isSindiceDcOrDcTerms(statement)) {
            return valueFactory.createStatement(statement.getSubject(),
                    toDcNamespace(statement.getPredicate(), valueFactory),
                    statement.getObject());
        } else {
            return statement;
        }
    }
```

File: metadata-transformer/src/main/java/no/unit/nva/metadata/service/MetadataService.java [hunk 20]
```diff
@@ -117,7 +155,8 @@ private JsonLdOptions avoidBlankNodeIdentifiersAndOmitDefaultsConfig() {
 
     private Map<String, Object> loadContext() {
         try {
-            var type = new TypeReference<Map<String,Object>>() {};
+            var type = new TypeReference<Map<String, Object>>() {
+            };
```
Comment (brinxmat) [inline metadata-transformer/src/main/java/no/unit/nva/metadata/service/MetadataService.java:159]:
```suggestion
            var type = new TypeReference<Map<String, Object>>() {};
```

**Review by brinxmat — CHANGES_REQUESTED** [review]

File: metadata-transformer/src/main/java/no/unit/nva/metadata/service/MetadataService.java [hunk 21]
```diff
@@ -73,23 +77,56 @@ private ByteArrayInputStream loadExtractedData() {
...
-            return QueryResults.asModel(repositoryConnection.getStatements(null, null, null));
+            try (RepositoryResult<Statement> statements = repositoryConnection.getStatements(null, null, null)) {
+                return fixUpSindiceDcAndDctermsToDcterms(statements);
+            }
         } finally {
             db.shutDown();
         }
     }
 
+    private Model fixUpSindiceDcAndDctermsToDcterms(RepositoryResult<Statement> statements) {
```
Comment (axthosarouris) [inline metadata-transformer/src/main/java/no/unit/nva/metadata/service/MetadataService.java:96]:
```suggest
private Model fixUpSindiceDcAndDctermsToDcterms(RepositoryResult<Statement> statements) {
        ValueFactory valueFactory = SimpleValueFactory.getInstance();
        Model model = new TreeModel();
        for (Statement statement : statements) {
            Statement transformedStatement = transformStatement(statement, valueFactory);
            model.add(transformedStatement);
        }
        return model;
    }

    private Statement transformStatement(Statement statement, ValueFactory valueFactory) {
        if (isSindiceDcOrDcTerms(statement)) {
            IRI statementPredicate = toDcNamespace(statement.getPredicate(), valueFactory);
            return valueFactory.createStatement(
                statement.getSubject(),
                statementPredicate,
                statement.getObject());
        } else {
            return statement;
        }
    }

    private boolean isSindiceDcOrDcTerms(Statement statement) {
        return statement.getPredicate().toString().toLowerCase(Locale.ROOT).startsWith(SINDICE_DC_URI_PART)
               || statement.getPredicate().toString().toLowerCase(Locale.ROOT).startsWith(SINDICE_DCTERMS_URI_PART);
    }
```

File: metadata-transformer/src/main/java/no/unit/nva/metadata/service/MetadataService.java (hunk 20 above)
Comment (axthosarouris) [inline metadata-transformer/src/main/java/no/unit/nva/metadata/service/MetadataService.java:159]: curly brackets in one line ?

File: metadata-transformer/src/main/java/no/unit/nva/metadata/service/MetadataService.java [hunk 22]
```diff
@@ -127,7 +166,6 @@ private JsonLdOptions avoidBlankNodeIdentifiersAndOmitDefaultsConfig() {
     private String getQueryAsString(URI uri) {
         var inputStream = inputStreamFromResources(MetadataService.QUERY_SPARQL);
         return new BufferedReader(new InputStreamReader(inputStream, StandardCharsets.UTF_8)).lines()
-                .collect(Collectors.joining(NEWLINE_DELIMITER)).replaceAll(REPLACEMENT_MARKER, uri.toString());
+            .collect(Collectors.joining(NEWLINE_DELIMITER)).replaceAll(REPLACEMENT_MARKER, uri.toString());
```
Comment (axthosarouris) [inline metadata-transformer/src/main/java/no/unit/nva/metadata/service/MetadataService.java:169]: Should NEWLINE_DELIMITER be set to System.lineSeparator()?

File: metadata-transformer/src/main/java/no/unit/nva/metadata/service/MetadataService.java (hunk 19 above)
Comment (mattiahj) [reply metadata-transformer/src/main/java/no/unit/nva/metadata/service/MetadataService.java:111]: Maybe we could use something more descriptive than "fix" in this method name?

File: metadata-transformer/src/main/java/no/unit/nva/metadata/service/MetadataService.java (hunk 22 above)
Comment (mattiahj) [reply metadata-transformer/src/main/java/no/unit/nva/metadata/service/MetadataService.java:169]: I believe so. Good find.

File: metadata-transformer/src/main/java/no/unit/nva/metadata/service/MetadataService.java (hunk 20 above)
Comment (mattiahj) [reply metadata-transformer/src/main/java/no/unit/nva/metadata/service/MetadataService.java:159]: Will fix.

File: metadata-transformer/src/test/java/no/unit/nva/metadata/service/MetadataServiceTest.java (hunk 15 above)
Comment (axthosarouris) [reply metadata-transformer/src/test/java/no/unit/nva/metadata/service/MetadataServiceTest.java:63]: i think the problem lies in somehow the parametrized test. It has become given a "special case input" assert that the parser parses it. Basically the definition of the test has become a parameter and that is confusing.

I think the tests have become a bit too complicated to read and grouped in an unexpected way. It would be more clear if we had tests that were 
1. this field maps to this field
2. The parser can handle upper-cased field values
3. The parser can handle lower-cased field values

Instead we have 
1. check lowercase abstract and lowercase title
2. check uppercase title and citation title.

My suggestion is : 
```
  @Test
    public  void dcTitleMapsToMainTitle() throws IOException {
        String title = "Title";
        CreatePublicationRequest request = createRequestWithTitle(title);
        Map<String, String> titleMap = Map.of(DC_TITLE_LOWERCASE, title);
        String html = createHtml(titleMap);
        assertThatParsedHtmlContainsExpectedMetadata(html, request);
    }

    @Test
    public  void lowerCasedFieldsMapCorrectly() throws IOException {
        String title = "Title";
        CreatePublicationRequest expectedRequest = createRequestWithLowerCaseFields(title);
        Map<String, String> lowerCaseFields = lowerCaseFields();
        String html = createHtml(lowerCaseFields);
        assertThatParsedHtmlContainsExpectedMetadata(html, expectedRequest);
    }

    private void assertThatParsedHtmlContainsExpectedMetadata(String html, CreatePublicationRequest expectedRequest)
        throws IOException {
        String filename = "article.html";
        URI uri = prepareWebServerAndReturnUriToMetadata(filename, html);
        MetadataService metadataService = new MetadataService();
        Optional<CreatePublicationRequest> request = metadataService.getCreatePublicationRequest(uri);

        assertThat(request.isPresent(), is(true));
        CreatePublicationRequest actual = request.get();
        actual.setContext(null);

        assertThat(actual, is(equalTo(expectedRequest)));
        
        stopServer();
    }

    @AfterEach
    public void stopServer() {
        wireMockServer.stop();
    }

```

Comment (axthosarouris) [reply metadata-transformer/src/test/java/no/unit/nva/metadata/service/MetadataServiceTest.java:63]: We do not have to fix this now though

File: metadata-transformer/src/main/java/no/unit/nva/metadata/service/MetadataService.java [hunk 23]
```diff
@@ -73,23 +75,54 @@ private ByteArrayInputStream loadExtractedData() {
...
-            return QueryResults.asModel(repositoryConnection.getStatements(null, null, null));
+            try (RepositoryResult<Statement> statements = repositoryConnection.getStatements(null, null, null)) {
+                return normalizeStatements(statements);
+            }
         } finally {
             db.shutDown();
         }
     }
 
+    private Model normalizeStatements(RepositoryResult<Statement> statements) {
```
Comment (mattiahj) [inline metadata-transformer/src/main/java/no/unit/nva/metadata/service/MetadataService.java:94]: More general, but maybe we want to normalize more stuff?

File: metadata-transformer/src/test/java/no/unit/nva/metadata/service/MetadataServiceTest.java (hunk 15 above)
Comment (mattiahj) [reply metadata-transformer/src/test/java/no/unit/nva/metadata/service/MetadataServiceTest.java:63]: Individual tests (conventional JUNIT use?) were brought up as a solution. I think we decided against it thinking there would be too much repeat of boilerplate, setup and teardown. The thinking being that this is more compact. However, I agree with you. It's not ideal. I feel like parametrized tests are nice, but should be used more to send varied, but mostly similar parameters into a single test, not as a way of handling setup and grouping of tests like we currently are.

Especially as we have more specific test cases then the ones you summarize, which is part of what prompted this change. This is a marked improvement though, we can revisit this again.

**Review by axthosarouris — APPROVED** [review]

File: metadata-transformer/src/main/java/no/unit/nva/metadata/service/MetadataService.java (hunk 23 above)
Comment (brinxmat) [reply metadata-transformer/src/main/java/no/unit/nva/metadata/service/MetadataService.java:95]: That was my thought too so +1

**Review by brinxmat — APPROVED** [review]
